lib/transport/GrpcTransport.js

```diff
@@ -0,0 +1,122 @@
+const GrpcErrorCodes = require('@dashevo/grpc-common/lib/server/error/GrpcErrorCodes');
+
+const MaxRetriesReachedError = require('./errors/MaxRetriesReachedError');
+const NoAvailableAddressesForRetry = require('./errors/NoAvailableAddressesForRetry');
+
+class GrpcTransport {
+  /**
+   * @param {createDAPIAddressProviderFromOptions} createDAPIAddressProviderFromOptions
+   * @param {
+   *    ListDAPIAddressProvider|
+   *    SimplifiedMasternodeListDAPIAddressProvider|
+   *    DAPIAddressProvider
+   * } dapiAddressProvider
+   * @param {DAPIClientOptions} globalOptions
+   */
+  constructor(createDAPIAddressProviderFromOptions, dapiAddressProvider, globalOptions) {
+    this.createDAPIAddressProviderFromOptions = createDAPIAddressProviderFromOptions;
+    this.dapiAddressProvider = dapiAddressProvider;
```

Comment:
Why do we need it here if we rewrite it with any request?

Reply:
Ah I see, if it's set then we don't

Reply:
Not with any, only in case if you pass it with options.